src/fund/hub/FundRouterMixin.sol

```diff
@@ -0,0 +1,33 @@
+pragma solidity 0.6.4;
+
+import "../../registry/IRegistry.sol";
+import "./IHub.sol";
+
+/// @title FundRouterMixin Contract
+/// @author Melon Council DAO <security@meloncoucil.io>
+/// @notice A mixin that provides routing for a given fund
+abstract contract FundRouterMixin {
+    function __getFeeManager(address _hub) internal view returns (address) {
```

Comment:
why not make these functions public instead so that both plugin and component contracts can use.

Reply:
@luongnt95 We could, but `public` functions take up much more space in bytecode than `internal` functions, because it creates interfaces to interact with the contract. Especially for fund contracts (which we deploy over and over again, per fund), we want to optimize space by only providing necessary external functions. (`public` is actually even worse than external in terms of efficiency, because it uses memory instead of calldata).

So in this case, we only provide the essential internal functions for consistent routing, which can then be implemented by other inheriting contracts (this is what policies and fees will do).

Remember that routes are already available via the hub, so these are purely convenience functions.